### PR TITLE
Tactical Speed & Damage Adjustments

### PR DESCRIPTION
**Pull request by Goosius1**
#### Description: 
This PR adds tactical speed and damage adjustments:
- Infantry encumbrance/slowing due to armour
  - Increases tactical options on the battlefield.
  - Makes room for the development of practical military uniforms.
  - Details:
```
  - Military Equipment reduces walking speeds:
    - Equipment:
      - Shield: -4 %
      - Full Leather Set: -4 %
      - Full Chainmail Set: -8 %
      - Full Iron Set: -16 %
      - Full Gold Set: -20 %
      - Full Diamond Set: -24 %    (*Walking around literally covered in rocks...*)
      - Full Netherite Set: -28 %
    - Players with 8% encumbrance or more are extra slow at jumping/climbing.
    - This feature increases tactical options on the battlefield.
    - This feature also allows nation to develop practically useful military uniforms.
      - Full diamond/netherite sets are no longer de-rigeur, due to their tactical vulnerability (*ie. reduced soldier speed*).
      - Instead, each nation can develop its own practically useful uniform, containing specific materials and (with leather) colours. 
```
- General movement increase +10% (to sweeten the above medicine).
- General damage decrease -30% (because players will generally have less armour than in vanilla).

____
#### New Nodes/Commands/ConfigOptions: 
```

############################################################
# +------------------------------------------------------+ #
# |                   SPEED ADJUSTMENTS                    | #
# +------------------------------------------------------+ #
############################################################
  
speed_adjustments:
  
  
  # +------------------------------------------------------+ #
  # |                      INFANTRY                        | #
  # +------------------------------------------------------+ #
  
  infantry:
  
  
    generic:
  
      # Adjusts the movement speed of all players.
      # TIP: Giving all players a small speed boost helps sugar-coat the concept of armour-based speed-slowing.
      adjustment_percentage: '10'
  
  
    armour_slowing:
  
      # If this value is true, armour slowing is enabled.
      # TIP 1: Armour-Slowing significantly boost the tactical options available to armies.
      # TIP 2: Armour-Slowing allows for the introduction of combat-useful national uniforms.
      # e.g. each nation may design its own national kit, including different materials, and (with leather items) colours.
      enabled: 'true'
  
      # The base slow percentage for each type of armour item.
      base_item_percentage:
  
        shield: '4'
  
        helmet: '0.6'
  
        chestplate: '1.6'
  
        leggings: '1.2'
  
        boots: '0.6'
  
      # The modification to slowing, of each material.
      material_modification_percentage:
  
        leather: '100'
  
        chainmail: '200'
  
        turtle_shell: '300'
  
        gold: '400'
  
        iron: '500'
  
        diamond: '600'
  
        netherite: '700'
  
  
############################################################
# +------------------------------------------------------+ #
# |                 DAMAGE ADJUSTMENTS                   | #
# +------------------------------------------------------+ #
############################################################
  
damage_adjustments:
  
  # This setting adjusts the damage of all weapons (including fists)
  # TIP: If armour slowing is enabled, this setting is important, as players will generally have less armour.
  all_weapons_percentage: '-30'
```
____
#### Relevant Issue ticket:
N/A

____
- [ x ] I have tested this pull request for defects on a server. 

By making this pull request, I represent that I have the right to waive copyright and related rights to my contribution, and agree that all copyright and related rights in my contributions are waived, and I acknowledge that the TownyAdvanced organization has the copyright to use and modify my contribution under the TownyCombat [License](https://github.com/TownyAdvanced/SiegeWar/blob/master/LICENSE.md) for perpetuity.

### CODE REVIEW

File: src/main/java/io/github/townyadvanced/townycombat/tasks/JumpReductionTask.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                    Towny.getPlugin().getServer().getScheduler().runTask(Towny.getPlugin(), new Runnable() {
+                        public void run() {
+                            player.addPotionEffect(new PotionEffect(PotionEffectType.SLOW, effectDurationTicks, 4, false, false, true));
+                            player.addPotionEffect(new PotionEffect(PotionEffectType.LEVITATION, effectDurationTicks, -30, false, false, true));
+                        }
+                    });
+                }
+            }
+        }
+    }
```
Comment (LlmDl) [inline src/main/java/io/github/townyadvanced/townycombat/tasks/JumpReductionTask.java:52]:
```suggestion
        if(armourSpeedAdjustmentPercentage != null && armourSpeedAdjustmentPercentage <= -8) {
            final int effectDurationTicks = (int)(20  + (((-armourSpeedAdjustmentPercentage / 4) -2) * 8));
           TownyCombat.getPlugin().getServer().getScheduler().runTask(TownyCombat.getPlugin(), ()-> applyEffect(player, effectDurationTicks));
       }
    }
	
    private void applyEffects(Player player, int effectDuractionTicks) {
        player.addPotionEffect(new PotionEffect(PotionEffectType.SLOW, effectDurationTicks, 4, false, false, true));
        player.addPotionEffect(new PotionEffect(PotionEffectType.LEVITATION, effectDurationTicks, -30, false, false, true));
    }
```

File: src/main/java/io/github/townyadvanced/townycombat/tasks/JumpReductionTask.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package io.github.townyadvanced.townycombat.tasks;
+
+import com.palmergames.bukkit.towny.Towny;
```
Comment (LlmDl) [inline src/main/java/io/github/townyadvanced/townycombat/tasks/JumpReductionTask.java:3]: The suggestion below will mean this import has to change.

**Review by LlmDl — COMMENTED** [review]
Looks good, I made one small suggestion which just makes things a bit simpler.

File: src/main/java/io/github/townyadvanced/townycombat/tasks/JumpReductionTask.java (hunk 1 above)
Comment (Goosius1) [reply src/main/java/io/github/townyadvanced/townycombat/tasks/JumpReductionTask.java:52]: Done.

File: src/main/java/io/github/townyadvanced/townycombat/tasks/JumpReductionTask.java (hunk 2 above)
Comment (Goosius1) [reply src/main/java/io/github/townyadvanced/townycombat/tasks/JumpReductionTask.java:3]: Done

**Review by LlmDl — APPROVED** [review]

**Pull request merged**